internal/code-generator/generator/documentation.go

```diff
@@ -2,13 +2,15 @@ package generator
 
 import (
 	"bytes"
-	"gopkg.in/yaml.v2"
-	"io"
 	"io/ioutil"
+
+	"gopkg.in/yaml.v2"
 )
 
-// Documentation reads from fileToParse and builds source file from tmplPath. It complies with type File signature.
-func Documentation(fileToParse, samplesDir, outPath, tmplPath, tmplName, commit string) error {
+// Documentation generates filename matchers in Go for documentation.
+// Based on template in tmplPath, using data from fileToParse and
+// result are saved to outFile. It is of generator.File type.
+func Documentation(fileToParse, _, outFile, tmplPath, tmplName, commit string) error {
```

Comment:
Is there some reason we need to keep the `samplesDir` path if the function doesn't use it? Or if there's a bunch of existing use, can we have a TODO pointing to an issue to fix it later?

Separately, this function has enough parameters that we might want to think about reworking it as a parameter struct at some point, but that's probably out of scope for this change.

Reply:
It's just to preserve the consistency: all generator functions implement `generator.File` (not sure if it's for a good reason, but they do), but this one was changed a bit to explicitly mark arg as not beeing used.

Generally, one thing to keep in mind is that this PR tries not to improve too much in the current state of the code (there are MANY opportunities) but only touches up on some low hanging fruits in very drive-by fashion